juju/juju
# provider/azure: use managed disks

## Description of change

This is a major change to the Azure provider,
to use managed disks. To use managed disks,
VMs must be placed in "aligned" availability
sets. Because of that, models support either
the old unmanaged disks or new managed disks,
but never both. Models created before this
change will continue to use unmanaged disks,
and new models will use managed disks.

The azure storage provider now has an optional
"account-type" attribute, which is the storage
account type: Standard_LRS or Premium_LRS.
The default is Standard_LRS. A defualt pool,
azure-premium, is added, with account-type set
to Premium_LRS.

For now, the model config's storage-account-type
only affects the storage type used for the root
disk of VMs in managed disk models. It continues
to affect all storage in unmanaged disk models.

## QA steps

1. juju bootstrap azure
2. juju deploy postgresql --storage pgdata=azure-premium,100G
3. juju deploy postgresql pg2 --storage pgdata=azure,100G
4. juju remove-unit pg2
5. juju add-unit pg2 --attach-storage=pgdata/1

Also, bootstrap with an older Juju, then upgrade with this branch, and check the old stuff still works, and new models work as they do with a freshly bootstrapped controller.

## Documentation changes

None.

## Bug reference

None.
